llvm/utils/update_mc_test_checks.py

```diff
@@ -118,6 +119,19 @@ def main():
         default=None,
         help="Set a default -march for when neither triple nor arch are found in a RUN line",
     )
+    parser.add_argument(
+        "--remove-duplicate",
+        action="store_true",
+        default=False,
+        help="remove duplicated test line if found",
+    )
+    parser.add_argument(
+        "--sort",
+        action="store_true",
+        default=False,
+        help="sort testline in alphabetic order (keep run-lines on top), this option could be dangerous as it"
+        "could change the order of lines that are not expected",
```

Comment:
What are these unexpected lines?

Reply:
The sorting is not very smart to handle all cases, and there are cases that might suprise the user, for example, the --sort and --remove-duplicate used at same time:

```
testlineA
//comment 1

testlineA
//commet 2
```
after remove-duplicate it become
```
testlineA
//comment 1

//commet 2
```

and then sort, `comment2` will be moved to other places which seems coming out of no where.

Reply:
This smells more like a bug rather than a limitation. We may report cases that only differ by their comments, but surely we shouldn't remove input lines from test cases and leave just comments?

Reply:
I think this might be hard to determine. i.e. the `//comment 2` can be a comment for the whole section bellow, and it should be kept even though the first testline is removed as a duplication. Thus I tried removing minimum lines when removing duplications and leave this to the user.

What do you think?

Reply:
> the `//comment 2` can be a comment for the whole section bellow,

If the script reported these duplicate cases (established by considering them with all comment lines stripped), then I think it shouldn't be a problem. The user will see the warning/error, fix the test case and run the script again.

I'd expect the script to play safe and only remove test cases when there's no doubt that they are duplicates, that is, when they are absolutely identical (which I guess should be the case for most of duplicates).

Reply:
updated